server/controllers/documents.js

```diff
@@ -0,0 +1,237 @@
+
+    /** convert limit and offset to number in base 10 */
+    limit = parseInt(req.query.limit, 10);
+    offset = parseInt(req.query.offset, 10);
+  }
+
+  Documents.findAll({
+    include: [
+      {
+        model: Users,
+        required: true,
+        attributes: ['userId', 'fullname'],
+      }
+    ],
+    offset,
+    limit,
+    attributes: { exclude: ['content', 'userId'] },
+  })
+  .then((documents) => {
+    if (documents.length < 1) {
+      return res.status(200).send({
+        status: 'ok',
+        message: 'No document found',
+      });
+    }
+
+    return res.status(200).send({
+      status: 'ok',
+      documents,
+    });
+  })
+  .catch(() => catchError(res));
+};
+
+/**
+ * @description view a document using its id
+ * @function getDocumentById
+ * @param {object} req request object
+ * @param {object} res response object
+ * @returns {object} response object
+ */
+const getDocumentById = (req, res) => {
+  req.checkParams('id', 'No document id supplied').notEmpty();
+  req.checkParams('id', 'Only integers are allowed as document id').isInt();
+
+  returnValidationErrors(req, res);
+
+  Documents.findOne({
+    where: { documentId: req.params.id },
+    attributes: { exclude: ['userId'] },
+    include: [
+      {
+        model: Users,
+        required: true,
+        attributes: ['userId', 'fullname', 'roleId'],
+        plain: true,
+      }
+    ],
+  })
+  .then((document) => {
+    if (!document) {
+      return res.status(404).send({
+        status: 'error',
+        message: 'This document does not exist or has been previously deleted',
+      });
+    }
+
+    if (
+      document.access === 'public' ||
+      req.decoded.role === 1 ||
+      document.User.userId === req.decoded.userId ||
+      (document.access === 'role' && document.User.roleId === req.decoded.role)
+    ) {
+      return res.status(200).send({
+        status: 'ok',
+        document,
+      });
+    }
+
+    return res.status(401).send({
+      status: 'error',
+      message: 'You are not authorized to view this document',
+    });
+  })
+  .catch(() => catchError(res));
+};
+
+/**
+ * @description updates a document
+ * @function updateDocument
+ * @param {object} req request object
+ * @param {object} res response object
+ * @returns {object} response object
+ */
+const updateDocument = (req, res) => {
+  req.checkParams('id', 'No document id supplied').notEmpty();
+  req.checkParams('id', 'Only integers are allowed as document id').isInt();
+
+  returnValidationErrors(req, res);
+
+  Documents.findOne({
+    where: { documentId: req.params.id },
+  })
+  .then((document) => {
+    if (!document) {
+      return res.status(404).send({
+        status: 'error',
+        message: 'This document does not exist or has been previously deleted',
+      });
+    }
+
+    /** allow updating if the decoded userId is same with the userId on the document */
+    if (req.decoded.userId === document.userId) {
+      document.update({
```

Comment:
***Code Quality: Cirtical/Nitpick***
Promises where made to solve the problem of call back hell. Using promises like this means you forgo the fact that you can use a more flattened nature of promises. which makes it hard for your code to be debuggable. This code could be refactored to be more elegant like so

```javascript
Documents.findOne({where: {documentId: req.params.id }.}).then((document) => {
    if(!document) // handle your error
    if(req.decoded.userId === document.userId) {
      return document.update({title: req.body.title}
}).then((doc) => your next steps)
```
The point is only have flat thens not nexted thens.
promises are monadic ish and they implement flat maps which takes care of promises of promises so you don't have to keep them deeply nexted